ndinterp/src/grid.rs

```diff
@@ -10,19 +10,25 @@
 //!     y = f(x1, x2, x3...)
 //!
 use crate::interpolate::InterpolationError;
-use ndarray::Array1;
+use itertools::izip;
+use ndarray::{Array, Dimension, Ix1};
 
 // Make public the families of interpolation algorithms implemented for grids
 pub mod cubic;
 
+/// A grid is made of two components:
+///     A d-dimensional vector of 1-dimensional sorted vectors for the input points
+///     A d-dimensional array for the grid values of
 #[derive(Debug)]
-pub struct Grid {
-    /// A grid is made of two (1-dimensional) sorted arrays.
-    pub input: Array1<f64>,
-    pub values: Array1<f64>,
+pub struct Grid<D: Dimension> {
+    /// Arrays with the input vectors
+    pub input: Vec<Vec<f64>>,
```

Comment:
Maybe I'd choose a different name here, like `points` or something else.

In the context of functions, `input` is really confusing...

Reply:
xgrid? (like in f(x1, x2, x3...))

Reply:
I thought about it as well, it just seemed redundant because the whole object is a `Grid`. But in lack of a better option is not bad.